### PR TITLE
monitoring: disable critical alerts banner by default

### PR DESCRIPTION
**Pull request by bobheadxi**
This banner is currently a significant source of confusion, and since we're still working on dogfooding alerts in sourcegraph.com (https://github.com/sourcegraph/sourcegraph/issues/5370) it's hard to say we're confident about all our alerts enough to have it displayed so prominently (yet)

This changes the default of `alerts.hideObservabilitySiteAlerts` to `true`

Most recent Slack discussion: https://sourcegraph.slack.com/archives/C07KZF47K/p1594730258245600

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -22,7 +22,7 @@ All notable changes to Sourcegraph are documented in this file.
     - PagerDuty notifiers: `severity` and `apiUrl`
     - Webhook notifiers: `bearerToken`
   - A new `disableSendResolved` option disables notifications for when alerts resolve themselves.
-- Recently firing critical alerts are now displayed to admins in site alerts.
+- Recently firing critical alerts can now displayed to admins via site alerts with the flag `alerts.hideObservabilitySiteAlerts` in user configuration.
```
[the remark this answers is not in the document]
Comment (bobheadxi) [reply CHANGELOG.md:25]: Thanks, that sounds a lot better 😅

File: schema/settings.schema.json [hunk 2]
```diff
@@ -175,7 +175,7 @@
     "alerts.hideObservabilitySiteAlerts": {
       "description": "Disables observability-related site alert banners.",
       "type": "boolean",
-      "default": false
+      "default": true
```
Comment (slimsag) [inline schema/settings.schema.json:178]: Sorry for the confusion, but these default values are not respected they only act as documentation. This had no impact on the actual default behavior - you need to handle that at the location where you use this setting in code

Comment (slimsag) [reply schema/settings.schema.json:178]: We're still seeing this on sourcegraph.com because of this 
image

Comment (slimsag) [reply schema/settings.schema.json:178]: When you fix this, make sure you cherry-pick the commit into the `3.18` branch so it gets into the release as well

Comment (bobheadxi) [reply schema/settings.schema.json:178]: Ah that's unfortunate - thanks for catching that! https://github.com/sourcegraph/sourcegraph/pull/12189

**Pull request merged**